pkg/target/host.go

```diff
@@ -148,12 +148,15 @@ func (h *Host) Flush(d time.Duration, updateHostHealthStatus chan *HostStatus) {
 			return
 		case <-t.C:
 			h.Wm.Lock()
-			if h.W != nil {
-				err := h.W.Flush()
-				if err != nil {
-					h.W = nil
-					h.Lg.Error("error while flushing the host buffer", zap.Error(err), zap.String("host name", h.Name), zap.Uint16("host port", h.Port))
-					h.closeUpdateHostConnection(updateHostHealthStatus)
+			if h.W != nil && h.Conn != nil {
+				if h.W.Buffered() != 0 {
+					err := h.W.Flush()
+					if err != nil {
+						h.W = nil
+						// TODO (grzkv) I am not sure this will nil connection fast enough
+						h.closeUpdateHostConnection(updateHostHealthStatus)
+						h.Lg.Error("error while flushing the host buffer", zap.Error(err), zap.String("host name", h.Name), zap.Uint16("host port", h.Port))
```

Comment:
can you add a metric for tracking this? would give better visibility into the overall picture.